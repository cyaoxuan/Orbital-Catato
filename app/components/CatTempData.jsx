// Temp data to showcase static pages

const catViewProps = [{
    iconName1: "information-circle",
    field1: "Status: ",
    iconName2: "location",
    field2: "Last Seen: "
}, {
    iconName1: "time",
    field1: "Last Fed: ",
    iconName2: "location",
    field2: "Last Seen: "
}];

const concernCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Engineering"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Skitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "New",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Utown"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Mitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Missing",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Science"
}, {
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Bitty",
    iconName1: catViewProps[0].iconName1,
    field1: catViewProps[0].field1,
    info1: "Injured",
    iconName2: catViewProps[0].iconName2,
    field2: catViewProps[0].field2,
    info2: "Arts"
}];

const fedCatCards = [{
    catImage: require("../../assets/temp-cat.jpg"),
    catName: "Kitty",
    iconName1: catViewProps[1].iconName1,
    field1: catViewProps[1].field1,
    info1: "19/05 12:01",
    iconName2: catViewProps[1].iconName2,
    field2: catViewProps[1].field2,
    info2: "Engineering"
}];

export { concernCatCards, fedCatCards }