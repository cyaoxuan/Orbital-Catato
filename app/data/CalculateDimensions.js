import { Dimensions } from "react-native";

// Get size of preview photos
export function getItemWidth(noOfItems, margin) {
    const { height, width } = Dimensions.get("window");
    // Width of screen - margins between images (8 * margin) / 4
    const itemSize = (width - margin * 2 * noOfItems) / noOfItems;
    return itemSize;
}
