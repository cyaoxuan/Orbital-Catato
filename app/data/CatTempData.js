// Temp data to showcase static pages
const cats = [{
    catID: 1,
    name: "Kitty",
    photoURLs: [require("../../assets/temp-cat.jpg"), require("../../assets/catato-logo.png")],
    gender: "M",
    birthYear: 2019,
    sterilised: true,
    keyFeatures: "Grey tabby and loves to sleep on his back",
    lastSeenLocation: "Engineering", // Will eventually be a geohash
    lastSeenTime: new Date(2023, 4, 19, 14, 21, 0, 0), // Will be converted from firebase timestamp
    lastFedTime: new Date(2023, 4, 19, 12, 1, 0, 0), 
    concernStatus: ["Injured"], 
    concernDesc: "Seen limping, possibly right front leg injured"
}, {
    catID: 2,
    name: "Skitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: null,
    birthYear: null,
    sterilised: false,
    keyFeatures: null,
    lastSeenLocation: "Utown",
    lastSeenTime: new Date(2023, 4, 20, 10, 53, 0, 0),
    lastFedTime: null,
    concernStatus: ["New", "Injured"],
    concernDesc: "New and had wounds on body"
}, {
    catID: 3,
    name: "Mitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: "F",
    birthYear: 2020,
    sterilised: true,
    keyFeatures: "White with orange patches with a heart-shaped patch on its back, tends to climb up to high places",
    lastSeenLocation: "Science",
    lastSeenTime: new Date(2023, 4, 15, 18, 34, 0, 0),
    lastFedTime: new Date(2023, 4, 15, 18, 34, 0, 0),
    concernStatus: ["Missing"], // The database registers as missing past certain time?
    concernDesc: "Has not been seen in 5 days"
}, {
    catID: 4,
    name: "Bitty",
    photoURLs: [require("../../assets/temp-cat.jpg")],
    gender: "M",
    birthYear: 2022,
    sterilised: true,
    keyFeatures: "Slightly chubby tabby and meows alot",
    lastSeenLocation: "Arts",
    lastSeenTime: new Date(2023, 4, 20, 9, 16, 0, 0),
    lastFedTime: new Date(2023, 4, 19, 20, 47, 0, 0),
    concernStatus: [], // If the array is empty it's healthy
    concernDesc: null
}];

export { cats }