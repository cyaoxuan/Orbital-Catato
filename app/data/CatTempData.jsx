// Temp data to showcase static pages
const cats = [{
    // For Select FlatList
    catID: 0,
    name: "New Cat",
    photoURL: require("../../assets/placeholder.png"),
    gender: null,
    birthYear: null,
    sterilised: null,
    keyFeatures: null,
    lastSeenLocation: null,
    lastSeenTime: null,
    lastFedTime: null,
    concernStatus: null,
    concernDesc: null
},  {
    catID: 1,
    name: "Kitty",
    photoURL: require("../../assets/temp-cat.jpg"), // Will eventually be an array of 
    gender: "M",
    birthYear: 2019,
    sterilised: true,
    keyFeatures: "Grey tabby and loves to sleep on his back",
    lastSeenLocation: "Engineering", // Will eventually be a geohash
    lastSeenTime: new Date(2023, 5, 19, 14, 21, 0, 0), // Will be converted from firebase timestamp
    lastFedTime: new Date(2023, 5, 19, 12, 1, 0, 0), 
    concernStatus: ["Injured"], 
    concernDesc: "Seen limping, possibly right front leg injured"
}, {
    catID: 2,
    name: "Skitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: null,
    birthYear: null,
    sterilised: false,
    keyFeatures: null,
    lastSeenLocation: "Utown",
    lastSeenTime: new Date(2023, 5, 20, 10, 53, 0, 0),
    lastFedTime: null,
    concernStatus: ["New", "Injured"],
    concernDesc: "New and had wounds on body"
}, {
    catID: 3,
    name: "Mitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: "F",
    birthYear: 2020,
    sterilised: true,
    keyFeatures: "White with orange patches with a heart-shaped patch on its back, tends to climb up to high places",
    lastSeenLocation: "Science",
    lastSeenTime: new Date(2023, 5, 15, 18, 34, 0, 0),
    lastFedTime: new Date(2023, 5, 15, 18, 34, 0, 0),
    concernStatus: ["Missing"], // The database registers as missing past certain time?
    concernDesc: "Has not been seen in 5 days"
}, {
    catID: 4,
    name: "Bitty",
    photoURL: require("../../assets/temp-cat.jpg"),
    gender: "M",
    birthYear: 2022,
    sterilised: true,
    keyFeatures: "Slightly chubby tabby and meows alot",
    lastSeenLocation: "Arts",
    lastSeenTime: new Date(2023, 5, 20, 9, 16, 0, 0),
    lastFedTime: new Date(2023, 5, 19, 20, 47, 0, 0),
    concernStatus: [], // If the array is empty it's healthy
    concernDesc: null
}];

export { cats }