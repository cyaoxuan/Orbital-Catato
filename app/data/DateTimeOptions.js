export const dateTimeOptions = {
    timeZone: "Asia/Singapore",
    month: "2-digit",
    day: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
    hour12: false
}