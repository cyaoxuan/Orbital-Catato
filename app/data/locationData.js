// JSON file format
// {
//     "User Location": {
//         "key": "1",
//         "value": "Use Current Location"
//     },
//     "The Deck": {
//         "key": "2",
//         "value": "The Deck",
//         "location": {"latitude": 1.2945075717691477, "longitude": 103.77256063897944},
//         "zone": "Arts"
//     },...
// }

const locationjson = require("./location.json");
const locations = Object.values(locationjson);
const zonejson = require("./zone.json");

export { locations, locationjson, zonejson };
