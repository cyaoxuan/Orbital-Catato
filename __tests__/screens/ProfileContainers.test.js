import { cleanup, fireEvent, render } from "@testing-library/react-native";
import { cats } from "../../app/data/CatTempData";
import { AvatarContainer, 
    KeyInfoContainer, 
    DetailsContainer, 
    PreviewPhotos, 
    PhotosContainer,
 } from "../../app/screens/main/catalogue/ProfileContainers";

afterEach(cleanup);

// Set time
beforeAll(() => {
    jest.useFakeTimers("modern");
    jest.setSystemTime(new Date(2023, 5, 20, 13, 12, 0, 0));
});

afterAll(() => {
    jest.useRealTimers();
});

describe("<AvatarContainer />", () => {
    it("renders avatar container successfully", () => {
        const avatarContainer = render(<AvatarContainer />);
        expect(avatarContainer).toBeDefined();
    });

    it("renders avatar container correctly", () => {
        const tree = render(<AvatarContainer />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<KeyInfoContainer />", () => {
    it("renders keyinfo container successfully", () => {
        const keyInfoContainer = render(<KeyInfoContainer cat={cats[0]} />);
        expect(keyInfoContainer).toBeDefined();
    });

    it("renders keyinfo container successfully", () => {
        const tree = render(<KeyInfoContainer cat={cats[0]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders keyinfo container with null cat fields correctly", () => {
        const { getAllByText } = render(<KeyInfoContainer cat={cats[4]}/>);
        const nullAgeGender = getAllByText("?");
        const falseSterilised = getAllByText("No");
        expect(nullAgeGender).toHaveLength(2);
        expect(falseSterilised).toHaveLength(1);
    });
});

describe("<DetailsContainer />", () => {
    it("renders details container successfully", () => {
        const detailsContainer = render(<DetailsContainer cat={cats[0]} variant="headlineSmall" />);
        expect(detailsContainer).toBeDefined();
    });

    it("renders details container correctly", () => {
        const tree = render(<DetailsContainer cat={cats[0]} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders details container with null cat fields correctly", () => {
        const { getAllByText } = render(<DetailsContainer cat={cats[4]}/>);
        const nullFeaturesTimes = getAllByText("Unknown");
        const emptyConcernArray = getAllByText("Healthy");
        const nullConcernsString = getAllByText("The cat is happy and healthy!");
        expect(nullFeaturesTimes).toHaveLength(3);
        expect(emptyConcernArray).toHaveLength(1);
        expect(nullConcernsString).toHaveLength(1);
    });
});

describe("<PreviewPhotos />", () => {
    it("renders preview photos successfully", () => {
        const photosContainer = render(<PreviewPhotos />);
        expect(photosContainer).toBeDefined();
    });

    it("renders preview photos without photoURLs", () => {
        const { getByText } = render(<PreviewPhotos />);
        const noPhotosText = getByText("This cat has no photos!");
        expect(noPhotosText).toBeDefined();
    });

    it("renders only the first 4 preview photos when photoURLs.length > 4", () => {
        const { getAllByTestId } = render(<PreviewPhotos photoURLs={cats[0].photoURLs} />);
        const previewPhotos = getAllByTestId("preview-photo");
        expect(previewPhotos).toHaveLength(4);
    });

    it("renders photos container correctly photoURLs.length < 4", () => {
        const tree = render(<PreviewPhotos photoURLs={cats[3].photoURLs} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<PhotosContainer />", () => {
    it("renders photos container successfully", () => {
        const photosContainer = render(<PhotosContainer />);
        expect(photosContainer).toBeDefined();
    });

    it("renders photos container correctly", () => {
        const tree = render(<PhotosContainer photoURLs={cats[0].photoURLs} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("calls onPress callback when touchable is pressed", () => {
        const onPressMock = jest.fn();
        const { getByTestId } = render(<PhotosContainer onPress={onPressMock} />);
        const viewTouchable = getByTestId("view-button");
        fireEvent.press(viewTouchable);
        expect(onPressMock).toHaveBeenCalled();
    });
});