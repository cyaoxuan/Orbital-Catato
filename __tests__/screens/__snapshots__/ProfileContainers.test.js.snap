// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarContainer /> renders avatar container correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderRadius": 100,
            "height": 200,
            "width": 200,
          },
          undefined,
        ]
      }
    >
      <Image
        accessibilityIgnoresInvertColors={true}
        source={1}
        style={
          {
            "borderRadius": 100,
            "height": 200,
            "width": 200,
          }
        }
        testID="avatar-image"
      />
    </View>
    <Text
      style={
        [
          {
            "fontFamily": "System",
            "fontSize": 32,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 40,
          },
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      Name
    </Text>
  </View>
</View>
`;

exports[`<DetailsContainer /> renders details container correctly 1`] = `
<View
  style={
    {
      "justifyContent": "space-evenly",
      "padding": 15,
    }
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "margin": 8,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 24,
          },
          {
            "marginHorizontal": 4,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon"
    >
      
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Key Features: 
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        Grey tabby and loves to sleep on his back
      </Text>
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "margin": 8,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 24,
          },
          {
            "marginHorizontal": 4,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon"
    >
      
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Last Seen: 
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        Engineering, 19/05, 22:21 (31d ago)
      </Text>
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "margin": 8,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 24,
          },
          {
            "marginHorizontal": 4,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon"
    >
      
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Last Fed: 
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        19/05, 20:01 (32d ago)
      </Text>
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "margin": 8,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 24,
          },
          {
            "marginHorizontal": 4,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon"
    >
      
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Concern Status: 
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        Injured
      </Text>
    </Text>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "margin": 8,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 24,
          },
          {
            "marginHorizontal": 4,
          },
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon"
    >
      
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "flex": 1,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Concerns: 
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            {
              "flex": 1,
            },
          ]
        }
      >
        Seen limping, possibly right front leg injured
      </Text>
    </Text>
  </View>
</View>
`;

exports[`<KeyInfoContainer /> renders keyinfo container successfully 1`] = `
<View
  style={
    {
      "backgroundColor": "lightgrey",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "padding": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Gender
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      M
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Age
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      4y
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Sterilised
    </Text>
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      Yes
    </Text>
  </View>
</View>
`;

exports[`<PhotosContainer /> renders photos container correctly 1`] = `
<View
  style={
    {
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      Photos
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="view-button"
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          View All
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": undefined,
                "fontSize": 12,
              },
              {
                "marginHorizontal": 4,
              },
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(202, 196, 208, 1)",
          "height": 0.5,
        },
        undefined,
        false,
        false,
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      testID="preview-photo"
    >
      <Image
        source={1}
        style={
          {
            "height": 171.5,
            "margin": 4,
            "resizeMode": "cover",
            "width": 171.5,
          }
        }
      />
    </View>
    <View
      testID="preview-photo"
    >
      <Image
        source={1}
        style={
          {
            "height": 171.5,
            "margin": 4,
            "resizeMode": "cover",
            "width": 171.5,
          }
        }
      />
    </View>
    <View
      testID="preview-photo"
    >
      <Image
        source={1}
        style={
          {
            "height": 171.5,
            "margin": 4,
            "resizeMode": "cover",
            "width": 171.5,
          }
        }
      />
    </View>
    <View
      testID="preview-photo"
    >
      <Image
        source={1}
        style={
          {
            "height": 171.5,
            "margin": 4,
            "resizeMode": "cover",
            "width": 171.5,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`<PreviewPhotos /> renders photos container correctly photoURLs.length < 4 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    testID="preview-photo"
  >
    <Image
      source={1}
      style={
        {
          "height": 171.5,
          "margin": 4,
          "resizeMode": "cover",
          "width": 171.5,
        }
      }
    />
  </View>
  <View
    testID="preview-photo"
  >
    <Image
      source={1}
      style={
        {
          "height": 171.5,
          "margin": 4,
          "resizeMode": "cover",
          "width": 171.5,
        }
      }
    />
  </View>
  <View
    testID="preview-photo"
  >
    <Image
      source={1}
      style={
        {
          "height": 171.5,
          "margin": 4,
          "resizeMode": "cover",
          "width": 171.5,
        }
      }
    />
  </View>
</View>
`;
